sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -490,7 +490,8 @@ object DDLPreprocessingUtils {
       case (expected, actual) =>
         if (expected.dataType.sameType(actual.dataType) &&
           expected.name == actual.name &&
-          expected.metadata == actual.metadata) {
+          expected.metadata == actual.metadata &&
+          expected.exprId.id == actual.exprId.id) {
```

Comment:
why does this fix the problem?

Reply:
This is not a correct change. please ignore this.